### PR TITLE
fix: Error response from persons API

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

A user was confused why the endpoint wasn't working with a distinctId. Now we have a better error message

## Changes

* Better error message
* Removed mixin that wasn't used anywhere else

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/api/person.py [hunk 1]
```diff
@@ -174,6 +175,24 @@ def get_queryset(self):
...
         return queryset
 
+    def get_object(self):
+        queryset = self.filter_queryset(self.get_queryset())
+        person_id = self.kwargs[self.lookup_field]
+
+        try:
+            queryset = get_pk_or_uuid(queryset, person_id)
+        except ValueError:
+            raise ValidationError(
```
Comment (pauldambra) [inline posthog/api/person.py:185]: Should this only be on certain actions?

E.g. if listing/retrieving then yes, but if updating it should be a 404?

Comment (benjackwhite) [reply posthog/api/person.py:185]: Hmmm not sure. If you are trying to call update but are passing in a dodgy looking ID I still think this error message is better than a 404 as we clearly know you're doing something wrong.

File: posthog/api/person.py [hunk 2]
```diff
@@ -174,6 +175,24 @@ def get_queryset(self):
...
+                f"The ID provided does not look like a personID. If you are using a distinctId, please use /persons?distinct_id={person_id} instead."
```
Comment (pauldambra) [inline posthog/api/person.py:186]:
```suggestion
                f"The ID provided does not look like a personID. If you are retrieving using a distinctId, please use /persons?distinct_id={person_id} instead."
```

Comment (pauldambra) [reply posthog/api/person.py:186]: Wandering into bike-shedding 😊

**Review by pauldambra — APPROVED** [review]
👍 yep, this comes up relatively frequently

File: posthog/api/person.py (hunk 2 above)
Comment (benjackwhite) [reply posthog/api/person.py:186]: Politely ignoring 😉

**Pull request merged**